src/compiler/checker.ts

```diff
@@ -3690,6 +3697,48 @@ module ts {
             }
             return links.resolvedType;
         }
+        
+        function getTypeGuardType(node: TypeGuardTypeNode): Type {
+            let type = <TypeGuardType>createType(TypeFlags.TypeGuard);
+            let hasParameters = false;
+            type.parameterIndex = -1;

```

Comment:
I'm not sure I understand? I'm using the parameterIndex right now to get the argument name from a call expression. How would a parameterSymbol help me?